include/rocksdb/listener.h

```diff
@@ -213,11 +213,29 @@ struct CompactionJobInfo {
   int base_input_level;
   // the output level of the compaction.
   int output_level;
-  // the names of the compaction input files.
+
+  // The following variables contain information about compaction inputs
+  // and outputs. A file may appear in both the input and output lists
+  // if it was simply moved to a different level. The order of elements
+  // is the same across input_files and input_levels_and_file_numbers;
+  // similarly, it is the same across output_files,
+  // output_levels_and_file_numbers, and output_oldest_blob_file_numbers.
+
+  // The names of the compaction input files.
   std::vector<std::string> input_files;
 
-  // the names of the compaction output files.
+  // The levels and file numbers of the compaction input files.
+  std::vector<std::pair<int, uint64_t>> input_levels_and_file_numbers;
+
+  // The names of the compaction output files.
   std::vector<std::string> output_files;
+
+  // The levels and file numbers of the compaction output files.
+  std::vector<std::pair<int, uint64_t>> output_levels_and_file_numbers;
+
+  // The oldest blob file referenced by each compaction output file.
+  std::vector<uint64_t> output_oldest_blob_file_numbers;
```

Comment:
It's may be my bias, but I don't particularly like the idea that two separate vectors and the element with the same index means the same file. Is there a way to consolidate `output_levels_and_file_numbers` and `output_oldest_blob_file_numbers` so that which blob file number is for which output file is clear? We can even consider to define another struct which contains level, file number and blob file number as members and have a vector of it here. We cannot touch `output_files` because we will have backward compatibility problem but if now we have a struct defined, in the future if we need to add anything new, we don't need to add yet another parameter here but just modify the struct there.

Reply:
Good point, will do.